### PR TITLE
fix: Do not return tools drawer in findDrawersTriggers test-util

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Preparing test utils for future changes

Related links, issue #, if available: n/a

### How has this been tested?

* PR build
* Did usage assessment, found no one relying on tools drawer to be present in that list

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__tests__/runtime-drawers.test.tsx [hunk 1]
```diff
@@ -53,29 +53,26 @@ describeEachAppLayout(({ size }) => {
   test('runtime drawers integration can be dynamically enabled and disabled', async () => {
     awsuiPlugins.appLayout.registerDrawer(drawerDefaults);
     const { wrapper, rerender } = await renderComponent(<AppLayout />);
-    // the 2nd trigger is for tools
-    expect(wrapper.findDrawersTriggers()).toHaveLength(2);
+    expect(wrapper.findDrawersTriggers()).toHaveLength(1);
```
Comment (just-boris) [inline src/app-layout/__tests__/runtime-drawers.test.tsx:56]: This behavior caused confusions before, so good to see this fixed

File: src/app-layout/__tests__/use-drawers.test.tsx [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+const someDrawers = [
+  { ...testDrawer, id: 'one' },
+  { ...testDrawer, id: 'two' },
+];
+
+function delay() {
+  return act(() => new Promise(resolve => setTimeout(resolve)));
+}
+
+function TestComponent({
```
Comment (just-boris) [inline src/app-layout/__tests__/use-drawers.test.tsx:21]: Testing all corner cases of this hook via real app layouts is tedious, so I decided to start this separate unit tests for drawers handling only

File: src/test-utils/dom/app-layout/index.ts [hunk 3]
```diff
@@ -62,7 +62,9 @@ export default class AppLayoutWrapper extends ComponentWrapper {
   }
 
   findDrawersTriggers(): ElementWrapper<HTMLButtonElement>[] {
-    return this.findAllByClassName<HTMLButtonElement>(testutilStyles['drawers-trigger']);
+    return this.findAll<HTMLButtonElement>(
+      `.${testutilStyles['drawers-trigger']}:not([data-testid="awsui-app-layout-trigger-awsui-internal-tools"])`
```
Comment (just-boris) [inline src/test-utils/dom/app-layout/index.ts:66]: There is a constant for this value: https://github.com/cloudscape-design/components/blob/e1c06524f6ed70ce60edf02cf1d2b8295db39dfd/src/app-layout/utils/use-drawers.ts#L13

But we cannot use it here because of CJS/ESM compatibility issues. This is why a separate value

**Comment by just-boris** [comment]
Changed the final plan, this is not needed
